plugin/build.gradle.kts

```diff
@@ -14,15 +14,22 @@ plugins {
     alias(libs.plugins.gradle.functional.test)
 }
 
+private val targetJvmRelease = provider { JavaLanguageVersion.of(11) }
+
 java {
-    targetCompatibility = JavaVersion.VERSION_11
-    sourceCompatibility = JavaVersion.VERSION_11
+    val javaVersion = targetJvmRelease.map { JavaVersion.toVersion(it.asInt()) }.get()
+    targetCompatibility = javaVersion
+    sourceCompatibility = javaVersion
+}
+
+tasks.withType<JavaCompile> {
+    options.release.set(targetJvmRelease.map { it.asInt() })
 }
 
 kotlin {
     compilerOptions {
         jvmTarget = JvmTarget.JVM_11
```

Comment:
question: why doesn't jvmTarget depend on current java? As I see Kotlin has JvmTarget for most of jvm version
https://github.com/JetBrains/kotlin/blob/master/libraries/tools/kotlin-gradle-compiler-types/src/generated/kotlin/org/jetbrains/kotlin/gradle/dsl/JvmTarget.kt

Reply:
Because jvmTarget for Kotlin must be the same as the one configured for Java. Otherwise, we get an exception during `compileKotlin` task